internal/iapl/policy.go

```diff
@@ -138,19 +163,48 @@ func (v *policy) validateUnions() error {
 				return fmt.Errorf("%s: resourceTypeNames: %s: %w", union.Name, rtName, ErrorUnknownType)
 			}
 		}
+
+		for _, rt := range union.ResourceTypes {
+			if _, ok := v.rt[rt.Name]; !ok {
+				return fmt.Errorf("%s: resourceTypes: %s: %w", union.Name, rt.Name, ErrorUnknownType)
+			}
+		}
 	}
 
 	return nil
 }
 
 func (v *policy) validateResourceTypes() error {
-	for _, resourceType := range v.p.ResourceTypes {
+	findRelationship := func(rels []Relationship, name string) bool {
+		for _, rel := range rels {
+			if rel.Relation == name {
+				return true
+			}
+		}
+
+		return false
+	}
```

Comment:
In general, if we have functions inside other functions, we should move them out unless there's a good reason (such as callback logic or some kind of closure over the environment). This doesn't seem like either of those, so it should probably be safe to move out.